aderyn_core/src/detect/experimental/closest_parent.rs

```diff
@@ -34,7 +34,7 @@ impl IssueDetector for ClosestParentDemonstrator {
                 capture!(self, context, block);
             }
 
-            if let Some(ASTNode::ForStatement(for_statement)) =
+            if let Some(for_statement) =
                 assignment.closest_parent_of_type(context, NodeType::ForStatement)
```

Comment:
example #2

Down below we can see `for_statement.closest_parent_of_type(context, NodeType::Block)`  but this time `for_statement` is of type `&ASTNode`